### DISCUSSION TITLE
Caching Data from S3

### DISCUSSION THREAD

**Discussion by ghost** [dragonflydb/dragonfly#2045] [Q&A] [answered]
On my production server (based on Axum, if that matters) I'm using S3 from Wasabi as a file database, works perfectly.
I would like to speedup file retrieval and reduce load on the storage by using my server to cache some data in memory and server from there.

Is there a way to make the DragonFlyDB kind of middle man to cache s3 files.

What kind of approach would you recommend?

**Comment by romange** [comment, marked as answer]
here is a nice article on how you can cache s3-like blobs in Dragonfly (anything that applies there applies to Dragonfly as well). 
https://aws.amazon.com/blogs/storage/turbocharge-amazon-s3-with-amazon-elasticache-for-redis/

I suggest setting up Dragonfly with "cache_mode=true" and set the cached objects with ttl to avoid stale data retrieval.
